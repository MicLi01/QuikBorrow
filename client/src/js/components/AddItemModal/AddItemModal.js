import React, { useState } from 'react'
import styled from 'styled-components'
import { Dialog, DialogTitle, DialogContent, DialogActions, FormHelperText, Typography, TextField, FormControl, IconButton } from '@material-ui/core'
import Dropzone from 'react-dropzone'
import teal from '@material-ui/core/colors/teal'
import { Button } from '@material-ui/core'
import ImageIcon from '@material-ui/icons/Image'
import CloseIcon from '@material-ui/icons/Close'

const FileSection = styled.section`
	margin-top: 10px;
`

const DropzoneStyles = styled.div`
	width: 100%;
	height: 100px;
	outline: ${teal[100]} dashed 3px;
	outline-offset: -4px;
	margin-top: 10px;
	display: flex;
	justify-content: center;
	align-items: center;
	margin-bottom: 10px;
  cursor: pointer;
  transition: all .3s;
  &:hover, &:active {
    outline: ${teal[500]} dashed 3px;
  }
`

const AddedFile = styled.div`
	display: flex;
  justify-content: space-between;
	align-items: center;
	height: 2em;
	background: ${teal[500]};
	padding: 4px 8px;
	color: ${teal[50]};
	overflow: hidden;
  margin-bottom: 4px;
  > div {
    display: flex;
    align-items: center;
  }
	p {
		margin-left: 10px;
		word-wrap: none;
		text-overflow: ellipsis;
		overflow: hidden;
	}
`


export function AddItemModal (props) {
  const { open, onClose } = props
  const [ files, setFiles ] = useState([])

  /**
   * @param {File[]} newFiles 
   */
  function addFiles(newFiles) {
    setFiles([ ...files, ...newFiles])
  }

  /**
   * @param {number} index
   */
  function removeFile(index) {
    files.splice(index, 1)
    setFiles([ ...files ])
  }

  return (
    <Dialog open={open} onClose={onClose} scroll='body'>
      <DialogTitle>
        Add an Item
      </DialogTitle>
      
      <DialogContent dividers>
        <Typography variant='body1'>
          Fill out some details, and we'll get you up and running ASAP!
        </Typography>
        <FormControl fullWidth margin='normal'>
          <TextField label='Maximum days' select required SelectProps={{ native: true }} margin='normal' fullWidth InputLabelProps={{ shrink: true }}>
            {[1,2,3,4,5,6,7].map(n => <option key={n} value={n}>{n}</option>)}
          </TextField>
          <FormHelperText>What's the maximum number of days you're willing to lend it out for?</FormHelperText>
          <TextField label='Item Name' required margin='normal' fullWidth InputLabelProps={{ shrink: true }} />
          <FormHelperText>You gotta have an item name, at the very least!</FormHelperText>
          <TextField label='Description' multiline rows='4' margin='normal' fullWidth InputLabelProps={{ shrink: true }}/>
          <FormHelperText>Give some more details about your item here</FormHelperText>
          
        </FormControl>
        <FileSection>
          <Typography variant='h5'>
            Images
          </Typography>
          <Dropzone onDrop={addFiles}>
            {({getRootProps, getInputProps}) => (
              <DropzoneStyles {...getRootProps()}>
                <input {...getInputProps()} />
                <Typography variant='subtitle1' color='textSecondary'>
                  Drag 'n' drop some files here, or click to select files
                </Typography>
              </DropzoneStyles>
            )}
          </Dropzone>
          {files.map((file, i) => (
            <AddedFile key={i+file.name+file.lastModified}>
              <div>
                <ImageIcon />
                <Typography variant='subtitle1' component='p'>
                {file.name}
                </Typography>
              </div>
              <IconButton aria-label='remove-image-upload' size='small' onClick={e => removeFile(i)}>
                <CloseIcon color='action' />
              </IconButton>
            </AddedFile>
          ))}
          
        </FileSection>
      </DialogContent>
      <DialogActions>
        <Button variant='text' onClick={onClose}>Cancel</Button>
        <Button color='primary' variant='contained'>Next</Button>
      </DialogActions>
    </Dialog>
  )
}

export default AddItemModal
