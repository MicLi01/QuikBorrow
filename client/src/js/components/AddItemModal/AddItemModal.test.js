import React from 'react';
import renderer from 'react-test-renderer';
import AddItemModal from './AddItemModal';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('AddItemModal', () => {
	test('renders', () => {
        const wrapper = mount(<AddItemModal open={true} onClose={jest.fn()} />)
        
		expect(wrapper.html()).toMatchSnapshot();
	});
});