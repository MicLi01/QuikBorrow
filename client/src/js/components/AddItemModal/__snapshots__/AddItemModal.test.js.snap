// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemModal renders 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollBody\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollBody MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogTitle-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Add an Item</h2></div><div class=\\"MuiDialogContent-root MuiDialogContent-dividers\\"><p class=\\"MuiTypography-root MuiTypography-body1\\">Fill out some details, and we'll get you up and running ASAP!</p><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required\\" data-shrink=\\"true\\">Maximum days<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><select class=\\"MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect\\" aria-invalid=\\"false\\" required=\\"\\"><option value=\\"1\\">1</option><option value=\\"2\\">2</option><option value=\\"3\\">3</option><option value=\\"4\\">4</option><option value=\\"5\\">5</option><option value=\\"6\\">6</option><option value=\\"7\\">7</option></select><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div><p class=\\"MuiFormHelperText-root\\">What's the maximum number of days you're willing to lend it out for?</p><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required\\" data-shrink=\\"true\\">Item Name<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div><p class=\\"MuiFormHelperText-root\\">You gotta have an item name, at the very least!</p><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink\\" data-shrink=\\"true\\">Description</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline\\"><textarea aria-invalid=\\"false\\" rows=\\"4\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\"></textarea></div></div><p class=\\"MuiFormHelperText-root\\">Give some more details about your item here</p></div><section class=\\"sc-bdVaJa eyOjMv\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\">Images</h5><div tabindex=\\"0\\" class=\\"sc-bwzfXH dSkuyU\\"><input multiple=\\"\\" type=\\"file\\" style=\\"display: none;\\" autocomplete=\\"off\\" tabindex=\\"-1\\"><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary\\">Drag 'n' drop some files here, or click to select files</h6></div></section></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Cancel</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Next</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
